internal/handler/msg_aggregate.go

```diff
@@ -248,10 +248,21 @@ func (h *Handler) MsgAggregate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
 	}
 
 	cancel := func() {}
+
+	timeout := time.NewTicker(time.Duration(maxTimeMS) * time.Millisecond)
```

Comment:
It's important to call `ticker.Stop()` to clear up resources when finished with a ticker.

Reply:
The PR is not ready yet for another round of reviews but thanks!